src/main/index.ts

```diff
@@ -5,362 +5,19 @@
-      "If you have HTTP_PROXY or http_proxy set in your environment, make sure that the localhost and the ipv4 loopback address 127.0.0.1 are added to the NO_PROXY environment variable.",
-    ];
-
-    dialog.showErrorBox("Lens Proxy Error", message.join("\n\n"));
-
-    return app.exit();
-  }
-
-  const extensionLoader = di.inject(extensionLoaderInjectable);
-
-  extensionLoader.init();
-
-  const extensionDiscovery = di.inject(extensionDiscoveryInjectable);
-
-  extensionDiscovery.init();
-
-  // Start the app without showing the main window when auto starting on login
-  // (On Windows and Linux, we get a flag. On MacOS, we get special API.)
-  const startHidden = process.argv.includes("--hidden") || (isMac && app.getLoginItemSettings().wasOpenedAsHidden);
-
-  logger.info("🖥️  Starting WindowManager");
-  const windowManager = di.inject(windowManagerInjectable);
-
-  const applicationMenuItems = di.inject(applicationMenuItemsInjectable);
-  const trayMenuItems = di.inject(trayMenuItemsInjectable);
-  const navigateToPreferences = di.inject(navigateToPreferencesInjectable);
-
-  onQuitCleanup.push(
-    initMenu(applicationMenuItems),
-    await initTray(windowManager, trayMenuItems, navigateToPreferences),
-    () => ShellSession.cleanup(),
-  );
-
-  installDeveloperTools();
-
-  if (!startHidden) {
-    windowManager.ensureMainWindow();
-  }
-
-  ipcMainOn(IpcRendererNavigationEvents.LOADED, async () => {
-    onCloseCleanup.push(startCatalogSyncToRenderer(catalogEntityRegistry));
-
-    const directoryForKubeConfigs = di.inject(directoryForKubeConfigsInjectable);
-
-    await ensureDir(directoryForKubeConfigs);
-
-    const kubeConfigSyncManager = di.inject(kubeconfigSyncManagerInjectable);
-
-    kubeConfigSyncManager.startSync();
-
-    startUpdateChecking();
-    lensProtocolRouterMain.rendererLoaded = true;
-  });
-
-  logger.info("🧩 Initializing extensions");
-
-  // call after windowManager to see splash earlier
-  try {
-    const extensions = await extensionDiscovery.load();
-
-    // Start watching after bundled extensions are loaded
-    extensionDiscovery.watchExtensions();
-
-    // Subscribe to extensions that are copied or deleted to/from the extensions folder
-    extensionDiscovery.events
-      .on("add", (extension: InstalledExtension) => {
-        extensionLoader.addExtension(extension);
-      })
-      .on("remove", (lensExtensionId: LensExtensionId) => {
-        extensionLoader.removeExtension(lensExtensionId);
-      });
-
-    extensionLoader.initExtensions(extensions);
-  } catch (error) {
-    dialog.showErrorBox("Lens Error", `Could not load extensions${error?.message ? `: ${error.message}` : ""}`);
-    console.error(error);
-    console.trace();
-  }
-
-  setTimeout(() => {
-    appEventBus.emit({ name: "service", action: "start" });
-  }, 1000);
-}
-
-main(getDi());
+(async () => {
+  await startApplication();
```

Comment:
> This await is useless

It is as so to make bigger deal about unawaited promise.


> Why not just have injecting startMainApplicationInjectable do all the work immediately?

It would be unexpected to have the act of inject do something operational while injected instance is irrelevant.